### PR TITLE
Add search index sort multi field

### PR DESCRIPTION
**Pull request by geichelberger**
The default sorting of the search index could be more intuitive. For example, lowercase letters follow uppercase letters, which leads to funny results that by sorting ascending, the letter `a` comes after `Z`.

By introducing a multi-field `sort` for selected fields, it is possible to ~normalize the string to lowercase, and with asciifolding~ use icu_collation_keyword field type which defaults to DUCET Collation a best effort attempt to better sort results.

https://www.elastic.co/guide/en/elasticsearch/plugins/current/analysis-icu-collation-keyword-field.html

An even better approach would be to use the `icu_collation` filter, which can be configured to use a collation table for a specific language.

Expected:
grafik

Current:
grafik

* [ ] have a concise title
* [ ] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [ ] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [ ] include migration scripts and documentation, if appropriate
* [ ] pass automated tests
* [ ] have a clean commit history
* [ ] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

### CODE REVIEW

**Comment by ziegenberg** [comment]
I would suggest going for the `icu_collation` filter which can be configured.

**Comment by gregorydlogan** [comment]
Considering that reindexing is already on the table for 16.0, is this going to be moved forward?

**Comment by geichelberger** [comment]
It's on my todo list for OC 16

**Comment by ziegenberg** [comment]
Installing the ICU plugin is not particularly difficult: `bin/opensearch-plugin install analysis-icu` :)

**Comment by geichelberger** [comment]
### Current vs ICU DUCET collation
grafik

File: docs/guides/admin/docs/installation/debs.md [hunk 1]
```diff
@@ -77,6 +77,10 @@ accomplished by running the following:
 If you wish to use the upstream OpenSearch repository directly be aware that Opencast only supported with OpenSearch 1.x
 and will not work with OpenSearch 2.x yet.  Future support for this is forthcoming.
 
+To install the ICU plugin for OpenSearch, run the following:
```
Comment (ziegenberg) [inline docs/guides/admin/docs/installation/debs.md]: We should tell the people why we need the ICU Plugin. Currently, this dependency is not justified in the docs.

**Review by ziegenberg — CHANGES_REQUESTED** [review]

**Comment by KatrinIhler** [comment]
The general tendency in the dev meeting was that making people re-index the whole thing instead of just the search part would be acceptable, especially since OC 16 is already a release with many breaking changes. So we probably don't want this to be configurable or only do the search index, but go all in.

File: docs/guides/admin/docs/installation/debs.md (hunk 1 above)
Comment (gregorydlogan) [reply docs/guides/admin/docs/installation/debs.md]: This should also clarify that things *will not work* without this call. I would absolutely skip this as a naive just-testing adopter, and then be confused when it all looks like it should work, but doesn't.

File: docs/guides/admin/docs/installation/rpm-el.md [hunk 2]
```diff
@@ -70,6 +70,8 @@ Make sure to install it on the node which also serves the admin interface.
 dnf install elasticsearch-oss
 ```
 
+
```
Comment (gregorydlogan) [inline docs/guides/admin/docs/installation/rpm-el.md:75]: Presumably the rpms need the same call made as the debs?

**Review by gregorydlogan — COMMENTED** [review]
Reminder that we need to incorporate this into #5597, or to incorporate that mapping file into this one, depending on who gets merged first.

**Comment by geichelberger** [comment]
It would be easy to update the index without a complete rebuild.

**Series Index Example:**

`PUT /opencast_series/_mapping`
``` json
{
  "properties": {
    "title": {
      "type": "keyword",
      "fields": {
        "sort": {
          "type": "icu_collation_keyword",
          "index": false
        }
      }
    },
    "description": {
      "type": "keyword",
      "fields": {
        "sort": {
          "type": "icu_collation_keyword",
          "index": false
        }
      }
    },
    "subject": {
      "type": "keyword",
      "fields": {
        "sort": {
          "type": "icu_collation_keyword",
          "index": false
        }
      }
    },
    "creator": {
      "type": "keyword",
      "fields": {
        "sort": {
          "type": "icu_collation_keyword",
          "index": false
        }
      }
    },
    "organizers": {
      "copy_to": "organizer",
      "type": "keyword",
      "fields": {
        "sort": {
          "type": "icu_collation_keyword",
          "index": false
        }
      }
    },
    "contributors": {
      "copy_to": "contributor",
      "type": "keyword",
      "fields": {
        "sort": {
          "type": "icu_collation_keyword",
          "index": false
        }
      }
    },
    "publisher": {
      "type": "keyword",
      "fields": {
        "sort": {
          "type": "icu_collation_keyword",
          "index": false
        }
      }
    },
    "rights_holder": {
      "type": "keyword",
      "fields": {
        "sort": {
          "type": "icu_collation_keyword",
          "index": false
        }
      }
    }
  }
}
```

Followed by:

` POST /opencast_series/_update_by_query?refresh&conflicts=proceed`

Update by query has many possible options. The version above is the simplest, and it may be enough, even if you have 200k entries, which are peanuts for elasticsearch.

**Comment by gregorydlogan** [comment]
This seems like something we should add to the docs as an option for those leery of doing a full rebuild!

**Comment by gregorydlogan** [comment]
@geichelberger this needs a note about the reindex, but then it should go into `develop` so it can be in for 16.0

**Comment by mtneug** [comment]
Is this still targeting 16?

File: docs/upgrade/15_to_16/update_event_mapping.json [hunk 3]
```diff
@@ -0,0 +1,83 @@
+{
```
Comment (gregorydlogan) [inline docs/upgrade/15_to_16/update_event_mapping.json]: These should be in 16_to_17 right?

**Review by gregorydlogan — APPROVED** [review]
Works as expected.  Should probably move the upgrade scripts, right?

File: docs/upgrade/15_to_16/update_event_mapping.json (hunk 3 above)
Comment (geichelberger) [reply docs/upgrade/15_to_16/update_event_mapping.json]: Yes, I moved the files.

**Comment by KatrinIhler** [comment]
FYI: Branch cut for OC 17 is on November 6!

**Pull request merged**